plugins/verifier/schemavalidator/schema_validator.go

```
@@ -0,0 +1,94 @@
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/deislabs/ratify/pkg/common"
+	"github.com/deislabs/ratify/pkg/ocispecs"
+	"github.com/deislabs/ratify/pkg/referrerstore"
+	_ "github.com/deislabs/ratify/pkg/referrerstore/oras"
+	"github.com/deislabs/ratify/pkg/verifier"
+	"github.com/deislabs/ratify/pkg/verifier/plugin/skel"
+	"github.com/deislabs/ratify/plugins/verifier/schemavalidator/schemavalidation"
+)
+
+type PluginConfig struct {
+	Name    string            `json:"name"`
+	Schemas map[string]string `json:"schemas"`
+}
+
+type PluginInputConfig struct {
+	Config PluginConfig `json:"config"`
+}
+
+func main() {
+	skel.PluginMain("schemavalidator", "1.0.0", VerifyReference, []string{"1.0.0"})
+}
+
+func parseInput(stdin []byte) (*PluginConfig, error) {
+	conf := PluginInputConfig{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse stdin for the input: %v", err)
+	}
+
+	return &conf.Config, nil
+}
+
+func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, referenceDescriptor ocispecs.ReferenceDescriptor, referrerStore referrerstore.ReferrerStore) (*verifier.VerifierResult, error) {
+	input, err := parseInput(args.StdinData)
+
+	if err != nil {
+		return nil, err
+	}
+
+	schemaMap := input.Schemas
+	ctx := context.Background()
+
+	referenceManifest, err := referrerStore.GetReferenceManifest(ctx, subjectReference, referenceDescriptor)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching reference manifest for subject: %s reference descriptor: %v", subjectReference, referenceDescriptor.Descriptor)
+	}
+
+	for _, blobDesc := range referenceManifest.Blobs {
+		refBlob, err := referrerStore.GetBlobContent(ctx, subjectReference, blobDesc.Digest)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching blob for subject:[%s] digest:[%s]", subjectReference, blobDesc.Digest)
+		}
+
+		err = processMediaType(schemaMap, blobDesc.MediaType, refBlob)
+		if err != nil {
```

Thanks for the updates Matt. Just want to confirm the expected behaviour here, if validation did not align with the schema, should we return an error? or return a VerifierResult where IsSuccess is false? if an error is returned, will the executor continue with other verifiers/artifacts?